lib/compress/zstd_compress.c

```diff
@@ -3469,9 +3469,12 @@ ZSTD_CDict* ZSTD_createCDict(const void* dict, size_t dictSize, int compressionL
 ZSTD_CDict* ZSTD_createCDict_byReference(const void* dict, size_t dictSize, int compressionLevel)
 {
     ZSTD_compressionParameters cParams = ZSTD_getCParams_internal(compressionLevel, ZSTD_CONTENTSIZE_UNKNOWN, dictSize);
-    return ZSTD_createCDict_advanced(dict, dictSize,
+    ZSTD_CDict* cdict = ZSTD_createCDict_advanced(dict, dictSize,
```

Comment:
minor : coding style :
prefer `ZSTD_CDict* const cdict = `
which indicates that, once created, the `cdict` pointer will never change.